### PR TITLE
loading graphs from files

### PR DESCRIPTION
**Pull request by AleksyBalazinski**
I haven't tested all the CLI options

### CODE REVIEW

File: GraphAlgorithms/Graphs/GraphLoader.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿namespace GraphAlgorithms.Graphs
+{
+    public class GraphLoader
+    {
+        private int currentLine;
+
+        private readonly string[] lines;
```
Comment (KarolDenst) [inline GraphAlgorithms/Graphs/GraphLoader.cs]: _lines

File: GraphAlgorithms/Graphs/GraphLoader.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        private GraphLoader(string[] lines)
+        {
+            this.lines = lines;
```
Comment (KarolDenst) [inline GraphAlgorithms/Graphs/GraphLoader.cs]: _lines

File: GraphAlgorithms/Graphs/GraphLoader.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                }
+
+                for (int j = 0; j < graphSize; j++)
+                {
+                    graph.AdjacencyMatrix[i, j] = values[j];
+                }
+            }
+
+            while (currentLine < lines.Length && lines[currentLine++].Trim().Length > 0)
+                ;
```
Comment (KarolDenst) [inline GraphAlgorithms/Graphs/GraphLoader.cs]: maybe split this into multiple lines

File: GraphAlgorithms/Program.cs [hunk 4]
```diff
@@ -27,10 +28,35 @@ private static void Main(string[] args)
         }, nameOption, algorithmTypeOption);
         rootCommand.Add(dimacsCommand);
 
-        var cliqueCommand = new Command("clique", "Find clique");
+        var cliqueCommand = new Command("max-clique", "Find maximum clique");
+        var fileOption = new Option<string>(name: "--file",
```
Comment (KarolDenst) [inline GraphAlgorithms/Program.cs:33]: maybe add options to command right after creating it or even cliqueCommand.Add(new Options...)
Tight now its difficult to make usre if all options are added

File: GraphAlgorithms/Program.cs [hunk 5]
```diff
@@ -27,10 +28,35 @@ private static void Main(string[] args)
...
+            .FromAmong("vertices", "vertices-then-edges");
+        cliqueCommand.Add(fileOption);
+        cliqueCommand.Add(indexOption);
+        cliqueCommand.Add(algorithmTypeOption);
+        cliqueCommand.Add(cmpOption);
+        cliqueCommand.SetHandler(RunCliqueFinder, fileOption, indexOption, algorithmTypeOption, cmpOption);
         rootCommand.Add(cliqueCommand);
 
-        var subgraphCommand = new Command("subgraph", "Find maximum common subgraph");
+        var subgraphCommand = new Command("mcs", "Find maximum common subgraph");
```
Comment (KarolDenst) [inline GraphAlgorithms/Program.cs]: also maybe seperate the creationg of comamnds into methods

File: GraphAlgorithms/Program.cs [hunk 6]
```diff
@@ -27,10 +28,35 @@ private static void Main(string[] args)
         }, nameOption, algorithmTypeOption);
         rootCommand.Add(dimacsCommand);
 
-        var cliqueCommand = new Command("clique", "Find clique");
+        var cliqueCommand = new Command("max-clique", "Find maximum clique");
+        var fileOption = new Option<string>(name: "--file",
+            description: "Path to file with graphs");
+        var indexOption = new Option<int>(name: "--index",
```
Comment (KarolDenst) [inline GraphAlgorithms/Program.cs:38]: all of this stuff should be required 
var indexOption = new Option<int>(name: "--index",
            description: "Index of the graph in the file",
            getDefaultValue: () => 0)
        {
            IsRequired = true
        };

**Review by KarolDenst — COMMENTED** [review]
The heuristic clique finder is bugging out. It adds 0 every time 
image
I thints because there are edges to itself. I think the graph loader should throw an exception if that is the case

File: GraphAlgorithms/Graphs/GraphLoader.cs (hunk 2 above)
Comment (AleksyBalazinski) [reply GraphAlgorithms/Graphs/GraphLoader.cs]: can you add .editorconfig? I tried to add one from VS but there's a bug preventing me from adding this on the solution level

File: GraphAlgorithms/Graphs/GraphLoader.cs (hunk 3 above)
Comment (AleksyBalazinski) [reply GraphAlgorithms/Graphs/GraphLoader.cs]: I also added an option to separate graphs by more than one blank line

File: GraphAlgorithms/Program.cs (hunk 6 above)
Comment (AleksyBalazinski) [reply GraphAlgorithms/Program.cs:38]: added required whenever there's no default.
also added aliases because this is becoming lengthy

**Review by KarolDenst — APPROVED** [review]

**Pull request merged**